modules/loader/src/main/java/com/opengamma/strata/loader/LoaderUtils.java

```diff
@@ -46,6 +99,178 @@ public static Index findIndex(String reference) {
 
   //-------------------------------------------------------------------------
   /**
+   * Parses a boolean from the input string.
+   * <p>
+   * Parsing is case insensitive.
+   * True is parsed as 'TRUE', 'T', 'YES', 'Y'.
+   * False is parsed as 'FALSE', 'F', 'NO', 'N'.
+   * Other strings are rejected.
+   * 
+   * @param str  the string to parse
+   * @return the parsed value
+   * @throws IllegalArgumentException if the string cannot be parsed
+   */
+  public static boolean parseBoolean(String str) {
+    switch (str.toUpperCase(Locale.ENGLISH)) {
+      case "TRUE":
+      case "T":
+      case "YES":
+      case "Y":
+        return true;
+      case "FALSE":
+      case "F":
+      case "NO":
+      case "N":
+        return false;
+      default:
+        throw new IllegalArgumentException("Unknown boolean value, must 'True' or 'False' but was '" + str + "'");
+    }
+  }
+
+  /**
+   * Parses a double from the input string.
+   * 
+   * @param str  the string to parse
+   * @return the parsed value
+   * @throws IllegalArgumentException if the string cannot be parsed
```

Comment:
It will actually throw `NumberFormatException`. It's a subclass of `IllegalArgumentException` so the comment isn't exactly wrong. But it could be more right.